### ISSUE TITLE
navigationBar doesn't display when we connect to reduxForm

### ISSUE THREAD

**Issue by heron2014** [DRDD2016/native#74]
Currently on Login and Sign up view we use ReduxForm, because of that we are unable to display navigationBar from ex-navigation. 

Question:
How to use default navigationBar and reduxForm?

code: 

```js
class Login extends React.Component {

  static route = {
    navigationBar: {
      title: 'Enter your new password',
      backgroundColor: 'blue',
      tintColor: 'white'
    }
  }

  render () {
    return (
      <View style={{ marginTop: 25, flex: 1 }}>
        <Button
          onPress={ () => { navigator.pop(); }}
        >
          <Icon name="chevron-left" size={ 28 } color="#525459" />
        </Button>
        <View style={{ marginTop: 50 }}>
          <Text style={{ paddingLeft: 5 }}>Email</Text>
          <View style={ styles.row }>
            <Field name="email" component={ FormTextInput } />
          </View>

          <Text style={{ paddingLeft: 5 }}>Password</Text>
          <View style={ styles.row }>
            <Field name="password" component={ FormPasswordInput } />
          </View>
          <View style={ styles.row }>
            <Button
              buttonStyle={{ backgroundColor: '#578de5', padding: 10, flex: 1, borderRadius: 5 }}
              textStyle={{ alignSelf: 'center', color: '#fff' }}
              onPress={this.props.handleSubmit(this.props.handleSubmitForm)}
            >
              <Text>LOGIN</Text>
            </Button>
          </View>
          <Text
            style={{ alignSelf: 'center' }}
            onPress={() => this.props.navigator.push('resetEmail')}
          >
            Forgot your password?
          </Text>
        </View>
      </View>
    );
  }
}


export default reduxForm({ form: 'login', validate })(Login);
```

```static property route``` is being ignored!

**Comment by heron2014** [comment]
Declaring static property outside the class:

```js
Login.route = {
  navigationBar: {
    title: 'Enter your new password',
    backgroundColor: 'blue',
    tintColor: 'white'
  }
}
```

doesn't work either

**Comment by heron2014** [comment]
I think the issue number 329 on ex-navigation repo can help us - 'static route not available to wrapped components'

**Comment by minaorangina** [comment]
Other examples of this problem:
- redux-form/issues/2230
- redux-form/issues/2237
- [Screenshots from someone's code](https://photos.google.com/share/AF1QipOoGmfv8MbfAXc9q0qThjXXIpyU65BkeYW1Ycg0L_aU0y-EyI9OIIxAim0pzst6vA?key=WUlELVFOVkpTYkpEb0VXdXQtSzJ1cVRJRkZUREpB)

**Comment by heron2014** [comment]
Solution:

```js
...
import { Field, reduxForm } from 'redux-form';
import hoistNonReactStatic from 'hoist-non-react-statics';
...

class Login extends Component {

  static route = {
    navigationBar: {
      title: 'Login',
      tintColor: colours.white,
      backgroundColor: colours.blue
    }
  }

  render () {
    const { .. } = this.props;

    return (
      <View style={{ flex: 1 }}>
       ...
      </View>
    );
  }
}

const decoratedComponent = reduxForm({ form: 'login', validate })(Login);
export default hoistNonReactStatic(decoratedComponent, Login);
```

**Comment by heron2014** [comment]
Resolved by #75
